### PR TITLE
Fix Hashing

### PR DESCRIPTION
**Pull request by jbspeakr**
#126 make hashing great again. 

The fix is mainly to instantiate the used MessageDigest for every single hashing operation. Applying this patch, we couldn't locally reproduce the current bug anymore. 

We decided to not fix all the problems in this app but to focus on a quick patch for the referenced issue.

### CODE REVIEW

**Comment by adetunjiakintundeakinde** [comment]
👍

**Comment by smirnov** [comment]
:+1:

**Comment by torwunder** [comment]
👍

File: src/main/java/org/zalando/planb/revocation/util/MessageHasher.java [hunk 1]
```diff
@@ -67,9 +68,8 @@ public String hashAndEncode(final RevocationType type, final String... messages)
...
 
         if (hashingAlgorithms().containsKey(type)) {
-
-            hashingAlgorithms().get(type).update((salt() + message).getBytes());
-            hashed = hashingAlgorithms().get(type).digest();
+            String algorithm = hashingAlgorithms().get(type);
+            hashed = MessageDigest.getInstance(algorithm).digest((salt() + message).getBytes());
         }
 
         return Base64.getUrlEncoder().encodeToString(hashed);
```
Comment (paul-zalando) [inline src/main/java/org/zalando/planb/revocation/util/MessageHasher.java]: only thing is the what encoding is used here? For the string I mean? If it varies between nodes then the resulting byte array will be different.

File: src/main/java/org/zalando/planb/revocation/api/ExceptionsResource.java [hunk 2]
```diff
@@ -142,8 +143,7 @@ public Problem accessDenied(final AccessDeniedException e) {
     @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
     @ResponseBody
     public Problem serializationException(final SerializationException e) {
-        log.error("A serialization exception occurred: {}", e.getMessage());
-        log.debug("Error details: ", e);
+        log.error("A serialization exception occurred", e);
```
Comment (paul-zalando) [inline src/main/java/org/zalando/planb/revocation/api/ExceptionsResource.java]: maybe only log like 4 or 5 stack frames? re-occuring error could really bog down the log

**Pull request merged**